sdk/core/azure-core/inc/credentials/credentials.hpp

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <chrono>
+#include <memory>
+#include <string>
+
+namespace azure
+{
+namespace core
+{
+namespace credentials
+{
+
+class Credential
+{
+public:
+  class _internal;
```

Comment:
But that's in global namespace?

Reply:
Technically 3.2 covers the global namespace, but we are recommending avoiding that prefix everywhere as shadowing is a common source of bugs. It would be annoying to run on some platform that happened to use that identifier and you only discover all the places you didn't qualify then.

Reply:
I removed the underscore, and named data members with `m_` prefix.